src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpRename.cs

```diff
@@ -577,6 +578,37 @@ void Goo()
         var y = 1;
         y = 2;
     }
+}");
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Rename)]
+        [WorkItem(39617, "https://github.com/dotnet/roslyn/issues/39617")]
+        public void VerifyRenameCaseChange()
+        {
+            var project = new ProjectUtils.Project(ProjectName);
+            VisualStudio.SolutionExplorer.AddFile(project, "Program.cs",
+@"
+class Program
+{
+    static void Main(string[] args)
+    {
+    }
+}");
+
+            VisualStudio.SolutionExplorer.OpenFile(project, "Program.cs");
+            VisualStudio.Editor.PlaceCaret("Program");
+
+            InlineRenameDialog.Invoke();
+
+            VisualStudio.Editor.SendKeys(VirtualKey.Home, VirtualKey.Delete, VirtualKey.P, VirtualKey.Enter);
+
+            VisualStudio.SolutionExplorer.Verify.FileContents(project, "program.cs",
```

Comment:
Does the Verify.FileContents _fail_ if the program.cs wasn't renamed at all and the file on disk is still called Program.cs?

Reply:
Hmmm.... I was really hoping it would check the file name matches case but it doesn't... It does verify that it's not called "Program1.cs" though =/